### PR TITLE
[CHAOS-234] Fix flaky cleanup for DNS disruption on restarted containers

### PR DESCRIPTION
**Pull request by nathantournant**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [X] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- Noticed during some tests a DNS disruption on containers that had restarted after injection cleanup was flaky
- After investigation, DNS disruption would not check for cgroup existence prior the write-to-reset attempt, making the injector crash and not consistently clean up.

## Code Quality Checklist

- [X] The documentation is up to date.
- [X] My code is sufficiently commented and passes continuous integration checks.
- [X] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [X] I leveraged continuous integration testing
    - [X] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [X] I manually tested the following steps:
    - Applying the tests descibed in the [Jira ticket]()
    - [X] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: injector/dns_disruption.go [hunk 1]
```diff
@@ -156,9 +156,16 @@ func (i DNSDisruptionInjector) Clean() error {
 	}
 
 	if i.config.Level == chaostypes.DisruptionLevelPod {
-		// reset net_cls classid
-		if err := i.config.Cgroup.Write("net_cls", "net_cls.classid", "0x0"); err != nil {
-			return fmt.Errorf("error writing classid to pod net_cls cgroup: %w", err)
+		// write default classid to pod net_cls cgroup if it still exists
+		exists, err := i.config.Cgroup.Exists("net_cls")
```
Comment (ptnapoleon) [inline injector/dns_disruption.go:160]: Should we add a similar check to the other injectors? I think the problem of "pod fails liveness probe, is restarted, container cgroup files reset" isn't specific to DNS?

Comment (nathantournant) [reply injector/dns_disruption.go:160]: It's specific to cgroups access, which is specific to DNS and Network disruptions
I made sure it's checked in both; I failed (8 months ago while writing a DNS bugfix) to copy/paste these exact lines from `network_disruption.go`
I hesitated to integrate/factorise this behavior into a new cgroup class function, and deemed it too specific for this class

Comment (nathantournant) [reply injector/dns_disruption.go:160]: Actually it's only consistently checked at cleanup, and consistently not at injection.
The risk we have is the container not existing -- and if a container doesn't exist the injection will fail much earlier

Comment (nathantournant) [reply injector/dns_disruption.go:160]: I have to test if the disruption is currently re-injected on container restart following the Dynamic Disruption change, but I don't think it is. Currently there's no mechanic to re-inject anything.
I'll create a low-priority feature ticket for this, this PR just aimed as fixing the cleanup issue.

**Review by ptnapoleon — APPROVED** [review]

**Pull request merged**